packages/application-generic/src/usecases/subscriber-job-bound/subscriber-job-bound.usecase.ts

```diff
@@ -0,0 +1,200 @@
+  StoreSubscriberJobsCommand,
+} from '../store-subscriber-jobs';
+import {
+  CreateNotificationJobs,
+  CreateNotificationJobsCommand,
+} from '../create-notification-jobs';
+import { PinoLogger } from 'nestjs-pino';
+import { ApiException } from '../../utils/exceptions';
+import {
+  ProcessSubscriber,
+  ProcessSubscriberCommand,
+} from '../process-subscriber';
+import { AnalyticsService } from '../../services/analytics.service';
+import { ProcessTenant } from '../process-tenant';
+import { SubscriberJobBoundCommand } from './subscriber-job-bound.command';
+
+const LOG_CONTEXT = 'SubscriberJobBoundUseCase';
+
+@Injectable()
+export class SubscriberJobBoundUsecase {
+  constructor(
+    private storeSubscriberJobs: StoreSubscriberJobs,
+    private createNotificationJobs: CreateNotificationJobs,
+    private processSubscriber: ProcessSubscriber,
+    private integrationRepository: IntegrationRepository,
+    private jobRepository: JobRepository,
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private processTenant: ProcessTenant,
+    private logger: PinoLogger,
+    private analyticsService: AnalyticsService
+  ) {}
+
+  @InstrumentUsecase()
+  async execute(command: SubscriberJobBoundCommand) {
+    this.logger.assign({
+      transactionId: command.transactionId,
+      environmentId: command.environmentId,
+      organizationId: command.organizationId,
+    });
+
+    const {
+      subscriber,
+      template,
+      templateProviderIds,
+      environmentId,
+      organizationId,
+      userId,
+      actor,
+      tenant,
+      identifier,
+    } = command;
+
+    await this.validateSubscriberIdProperty(subscriber);
+
+    this.analyticsService.track(
+      'Notification event trigger - [Triggers]',
+      command.userId,
+      {
+        transactionId: command.transactionId,
+        _template: template._id,
+        _organization: command.organizationId,
+        channels: template?.steps.map((step) => step.template?.type),
+        source: command.payload.__source || 'api',
+      }
+    );
+
+    const subscriberProcessed = await this.processSubscriber.execute(
+      ProcessSubscriberCommand.create({
+        environmentId,
+        organizationId,
+        userId,
+        subscriber,
+      })
+    );
+
+    // If no subscriber makes no sense to try to create notification
+    if (!subscriberProcessed) {
+      /**
+       * TODO: Potentially add a CreateExecutionDetails entry. Right now we
+       * have the limitation we need a job to be created for that. Here there
+       * is no job at this point.
+       */
+      Logger.warn(
+        `Subscriber ${JSON.stringify(
+          subscriber.subscriberId
+        )} of organization ${command.organizationId} in transaction ${
+          command.transactionId
+        } was not processed. No jobs are created.`,
+        LOG_CONTEXT
+      );
+
+      return;
+    }
+
+    const notificationJobs = await this.createNotificationJobs.execute(
```

Comment:
Now I understand the naming a bit but we might consider if the main goal of the use case is to create the jobs needed and processing the subscriber is the dependency as we need their information to proper create the jobs. Mention that as it seems by the naming that the main goal of the new queue is to process subscribers but at the end what it does is to create all the jobs needed for them through AddJob to add them to a different queue.